test/test.js

```diff
@@ -235,6 +239,33 @@ describe('Page', function() {
     }));
   });
 
+  describe('Browser.Events.disconnected', function() {
+    it('should emitted when browser gets closed', SX(async function() {
+      const browser = await puppeteer.launch(defaultBrowserOptions);
+      let isDisconnected = false;
+      browser.on('disconnected', dc => {
+        expect(dc).toBe('connection disconnected');
+        isDisconnected = true;
+      });
+      await browser.close();
+      expect(isDisconnected).toBe(true);
+    }));
+    it('should emitted when browser.disconnect() called', SX(async function() {
+      const originalBrowser = await puppeteer.launch(defaultBrowserOptions);
+      const newBrowser = await puppeteer.connect({
+        browserWSEndpoint: originalBrowser.wsEndpoint()
+      });
+      let isDisconnected = false;
+      newBrowser.on('disconnected', dc => {
+        isDisconnected = true;
+      });
+      newBrowser.disconnect();
+      expect(isDisconnected).toBe(true);
+    }));
+    xit('should be emitted when underlying websocket gets closed', SX(async function() {
```

Comment:
you should be able to:
1. launch a browser
2. connect to it with `pptr.connect`
2. close the browser, make sure the connected one issues the `disconnected` event

Reply:
Let me know if it still not fit